Remove codegen from supplementary_tests

bjoernschroeder opened the pull request:
Based on https://github.com/rapyuta-robotics/alica/pull/410

bjoernschroeder committed:
move behaviours into libalica-tests, use dynamic loading

bjoernschroeder committed:
Merge branch 'devel' into bas_use_dynamic_loading_behaviours

bjoernschroeder committed:
add dynamic loading for TestBehaviour

bjoernschroeder committed:
remove libalica-tests from dependencies

bjoernschroeder committed:
use dynamic loading for behaviours in supplementary_tests

bjoernschroeder committed:
Merge branch 'devel' into bas_use_dynamic_loading_behaviours

bjoernschroeder committed:
remove old line, fix typo

bjoernschroeder committed:
set Rapyuta Robotics as author / maintainer for libs

bjoernschroeder committed:
remove package files for libs

bjoernschroeder committed:
move add_subdirectory out of if block

bjoernschroeder committed:
use dynamic loading for all alica elements in supplementary_tests

bjoernschroeder committed:
remove preConditionConstraints, move VariableHandlingStart transition condition to lib, fix dynamic loading test

bjoernschroeder committed:
format

bjoernschroeder committed:
Merge branch 'devel' into bas_remove_codegen

Dale-Koenig commented on supplementary/alica_ros1/supplementary_tests/src/test/etc/plans/conditions/ConditionRepository.cnd:
```diff
@@ -1,26 +1,64 @@
+      "comment": "AnyChildSuccess",
+      "conditionRepository": 0,
+      "id": 1,
+      "libraryName": "libalica-util",
```
This library doesn't exist anymore.  Are these conditions used at all?

Dale-Koenig commented on supplementary/alica_ros1/alica_tests/libalica-tests/CMakeLists.txt:
```diff
@@ -0,0 +1,43 @@
+cmake_minimum_required(VERSION 3.5.1)
+project(libalica-tests)
```
Can we combine with alica_tests main Cmakelists file?

Dale-Koenig replied:
ping here

Dale-Koenig commented on supplementary/alica_ros1/supplementary_tests/libalica-supplementary-tests/include/libalica-supplementary-tests/VariableHandling/VHMaster.h:
```diff
@@ -0,0 +1,51 @@
+
+#include <boost/dll/alias.hpp>
+
+extern bool vhStartCondition;
```
This is weird.  Is this new?  What problem is this trying to solve?

bjoernschroeder replied:
Its a boolean value which at some point in one of the tests will be set from false to true to trigger a transition and is a leftover from old code. I can replace it by using the global blackboard instead and remove it.

veerajsk commented on supplementary/alica_ros1/supplementary_tests/libalica-supplementary-tests/include/libalica-supplementary-tests/GSolver/GSolverTestPlan.h:
```diff
@@ -0,0 +1,47 @@
+class GSolverTestPlanUtilityFunction : public BasicUtilityFunction
+{
+public:
+    std::shared_ptr<UtilityFunction> getUtilityFunction(Plan* plan);
```
you can reduce a lot of the repeated code here by using base classes & templates as done in lbc. Since you have already done the changes though, its okay, but you can use this method in other PRs

ntfshard commented on supplementary/alica_ros1/supplementary_tests/src/test/test_alica_dynamicloadcreator.cpp:
```diff
@@ -132,17 +132,24 @@ TEST_F(AlicaDynamicLoading, simple_transition_condition_load)
     std::string path = getRootPath();
 
     YAML::Node node;
+    YAML::Node condition;
```
variable scope can be reduced

ntfshard commented on supplementary/alica_ros1/supplementary_tests/libalica-supplementary-tests/src/libalica-supplementary-tests/ProblemModule/QueryPlan2.cpp:
```diff
@@ -0,0 +1,27 @@
+        : BasicPlan(context)
+{
+}
+QueryPlan2::~QueryPlan2() {}
```
redundant ?

ntfshard commented on supplementary/alica_ros1/supplementary_tests/libalica-supplementary-tests/include/libalica-supplementary-tests/GSolver/GSolverMaster.h:
```diff
@@ -0,0 +1,29 @@
+{
+public:
+    GSolverMaster(PlanContext& context);
+    virtual ~GSolverMaster();
```
virtual -> override
same below

and if it's empty you can omit it

ntfshard commented on supplementary/alica_ros1/supplementary_tests/libalica-supplementary-tests/src/libalica-supplementary-tests/ProblemModule/ProblemBuildingMaster.cpp:
```diff
@@ -0,0 +1,31 @@
+#include <assert.h>
```
unused

bjoernschroeder committed:
- remove unused includes
- use standard library for conditions
- replace extern bool vhStartCondition with globalBlackboard
- remove empty destructors of alica elements

bjoernschroeder committed:
remove cmake files for libaries, remove unused condition file in supplementary_tests

bjoernschroeder committed:
remove expr folder from supplementary_tests

Dale-Koenig committed:
Merge branch 'devel' into bas_remove_codegen
